feat: replacement support for all managers (#17883)
renovatebot/renovate: docs/usage/configuration-options.md

### replacementName

> This config option is not working with all managers.

Reviewer: ```suggestion
This config option only works on some managers.
```

We're working to support more managers, subscribe to issue [renovatebot/renovate#14149](https://github.com/renovatebot/renovate/issues/14149) to follow our progress.

> Managers which currently do not support replacement:

Reviewer: ```suggestion
Managers which do not support replacement:
```

> - bazel
> - git-submodules
> - gomod
> - gradle
> - hermit
> - homebrew
> - maven
> - regex

Reviewer: ```suggestion
- `bazel`
- `git-submodules`
- `gomod`
- `gradle`
- `hermit`
- `homebrew`
- `maven`
- `regex`
```

> Use this field to define the name of a replacement package.

Reviewer: ```suggestion
Use the `replacementName` config option to set the name of a replacement package.
```

Must be used with `replacementVersion` (see example below).
You can suggest a new community package rule by editing [the `replacements.ts` file on the Renovate repository](https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/replacements.ts) and opening a pull request.

### replacementVersion

> This config option is not working with all managers.

Reviewer: ```suggestion
This config option only works with some managers.
```

We're working to support more managers, subscribe to issue [renovatebot/renovate#14149](https://github.com/renovatebot/renovate/issues/14149) to follow our progress.

> Managers which currently do not support replacement:

Reviewer: ```suggestion
Managers which do not support replacement:
```

> - bazel
> - git-submodules
> - gomod
> - gradle
> - hermit
> - homebrew
> - maven
> - regex

Reviewer: ```suggestion
- `bazel`
- `git-submodules`
- `gomod`
- `gradle`
- `hermit`
- `homebrew`
- `maven`
- `regex`
```

> - regex

Reviewer 2: Can we avoid duplicating this list? Is it enough to list on `replacementName`?

Reviewer 3: @HonkingGoose thoughts on the above question?

Reviewer: We should have a sentence like this in `replacementVersion` docs, and remove the list of managers from `replacementVersion`.

> For a list of managers which do not support replacement read the `replacementName` config option docs.

> Use this field to define the version of a replacement package.

Reviewer: ```suggestion
Use the `replacementVersion` config option to set the version of a replacement package.
```
